hudi-common/src/main/java/org/apache/hudi/common/util/collection/DiskBasedMap.java

```diff
@@ -395,4 +417,48 @@ public int compareTo(ValueMetadata o) {
       return Long.compare(this.offsetOfValue, o.offsetOfValue);
     }
   }
+
+  private static class DiskCompressionInstance implements Serializable {
+    public static final int DISK_COMPRESSION_INITIAL_BUFFER_SIZE = 1048576;
+
+    // Caching ByteArrayOutputStreams to avoid recreating it for every operation
+    private final ByteArrayOutputStream compressBaos;
+    private final ByteArrayOutputStream decompressBaos;
+    private final byte[] decompressBuffer;
+
+    DiskCompressionInstance() {
+      compressBaos = new ByteArrayOutputStream(DISK_COMPRESSION_INITIAL_BUFFER_SIZE);
+      decompressBaos = new ByteArrayOutputStream(DISK_COMPRESSION_INITIAL_BUFFER_SIZE);
+      decompressBuffer = new byte[8192];
+    }
+
+    public byte[] compressBytes(final byte [] value) throws IOException {
```

Comment:
I assume these are not required to be thread safe ? can you confirm this.

Reply:
are these required to be public ?

Reply:
Yeah this is optimized for running as a single thread.